### PR TITLE
feat: rework `rawSlice` to support padding and correctly handle overflows

### LINKED ISSUE #796: `rawSlice()` should support bit padding
This is a follow-up on #787 

String literal type `s` in FunC supports bit padding.
I'm sure that we should also support that for Tact's `rawSlice()`, because otherwise it won't be possible to define slices with binary data of length not divisible by 4.

This padding basically removes all the trailing zeros and also the last `1` bit before them. So, for example, `4a` is `0100 1010` and `4a_` is `0100 10`.

### PR DESCRIPTION
**Pull request by Gusarich**
## Issue

Closes #796.

## Checklist

- [x] I have updated CHANGELOG.md
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Review by novusnota — DISMISSED** [review]

File: src/test/e2e-emulated/contracts/intrinsics.tact
Comment (anton-trunov) [inline src/test/e2e-emulated/contracts/intrinsics.tact:1]: Let's add a couple of tests groups from the legendary `tvm.pdf`:

> For example, 8A corresponds to binary string 10001010, while 8A_ and 8A0_ both correspond to 100010.

> An empty bitstring may be represented by either ‘’, ‘8_’, ‘0_’, ‘_’, or ‘00_’.

File: src/interpreter.ts [hunk 1]
```diff
@@ -1096,30 +1097,60 @@ export class Interpreter {
...
+                        const paddedBits = paddedBufferToBits(buffer);
+
+                        console.log({
+                            originalHex: hex,
+                            paddedHex,
+                            bufferLength: buffer.length,
+                            paddedBitsLength: paddedBits.length,
+                            bitStringLength: bits.length,
+                            offset: hex.length % 2 === 0 ? 0 : 4,
+                        });
```
Comment (anton-trunov) [inline src/interpreter.ts]: Is it some left-over from debugging?

File: src/interpreter.ts [hunk 2]
```diff
@@ -1096,30 +1097,60 @@ export class Interpreter {
...
 
-                    try {
-                        return beginCell()
-                            .storeBuffer(Buffer.from(str, "hex"))
-                            .endCell()
-                            .asSlice();
-                    } catch (_) {
+                    // Remove underscores from the hex string
+                    const hex = str.replace("_", "");
+                    const paddedHex = hex.length % 2 === 0 ? hex : "0" + hex;
```
Comment (anton-trunov) [inline src/interpreter.ts:1109]: We need a test that justifies this `paddedHex` thing. AFAIU, if the padding is removed the tests won't catch it.

**Review by anton-trunov — CHANGES_REQUESTED** [review]

File: src/test/e2e-emulated/contracts/intrinsics.tact
Comment (Gusarich) [reply src/test/e2e-emulated/contracts/intrinsics.tact:1]: done

File: src/interpreter.ts (hunk 1 above)
Comment (Gusarich) [reply src/interpreter.ts]: removed

File: src/interpreter.ts (hunk 2 above)
Comment (Gusarich) [reply src/interpreter.ts:1109]: there are multiple test cases with an odd number of digits already: `0_`, `1_`, `7`

Comment (anton-trunov) [reply src/interpreter.ts:1109]: got it, thanks

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**